api/lib/domain/models/Student.js

```diff
@@ -0,0 +1,18 @@
+const _ = require('lodash');
+
+class Student {
+
+  constructor({
+    nationalStudentId,
+    accounts = [],
+  } = {}) {
+    this.nationalStudentId = nationalStudentId;
+    this.accounts = accounts;
+  }
+
+  getAppropriateAccount() {
+    return _.orderBy(this.accounts, ['certificationCount', 'updatedAt'], ['desc', 'desc'])[0];
+  }
```

Comment:
Z'êtes prêts ?
```suggestion
  accountWithMostCertificationsIfEqualsMostRecent() {
    return _.orderBy(this.accounts, ['certificationCount', 'updatedAt'], ['desc', 'desc'])[0];
  }

  mostRelevantAccountForReconciliation() {
    return this.accountWithMostCertificationsIfEqualsMostRecent();
  }
```


En contexte dans   api/lib/infrastructure/repositories/schooling-registration-repository.js
```
            if (student) {
          //  schoolingRegistration.userId = student.getAppropriateAccount().userId;
              schoolingRegistration.userId = student.mostRelevantAccountForReconciliation().userId;
            }
```

Reply:
C'est une illustration [de ça ](https://twitter.com/tdpauw/status/1172151176114188291)

Reply:
La méthode a été renommée en `getMostRelevantAccountForReconciliation()` mais je ne suis pas particulièrement partant pour avoir une deuxième méthode qui explicite la règle métier.